lib/E2EEnabledPathCache.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+// SPDX-FileCopyrightText: 2020 Georg Ehrke <georg-nextcloud@ehrke.email>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+namespace OCA\EndToEndEncryption;
+
+use Sabre\DAV\INode;
+use OCP\Files\Node;
+use OCP\Files\IHomeStorage;
+use OCP\Files\Cache\ICache;
+use OC\Files\Storage\Wrapper\Wrapper;
+use OCA\Files_Sharing\SharedStorage;
+
+class E2EEnabledPathCache {
+	/**
+	 * @psalm-type FileId=int
+	 *
+	 * @psalm-type EncryptedState=array{0: FileId, 1: bool}
+	 *
+	 * @psalm-type Path=string
+	 *
+	 * @psalm-type StorageId=string|int
+	 */
+
+	/** @var array<StorageId, array<Path, EncryptedState>> */
+	protected $cacheEntries;
```

Comment:
rename to perStorageEncryptedStateCache ? :sweat_smile: 

the generic cacheEntries name doesn't really evoke its structure